internal/cmd/login.go

```diff
@@ -25,57 +24,48 @@ import (
 	"github.com/infrahq/infra/internal/cmd/cliopts"
 	"github.com/infrahq/infra/internal/cmd/types"
 	"github.com/infrahq/infra/internal/format"
-	"github.com/infrahq/infra/internal/generate"
 	"github.com/infrahq/infra/internal/logging"
 )
 
 type loginCmdOptions struct {
 	Server             string
 	AccessKey          string
-	Provider           string
 	SkipTLSVerify      bool
 	TrustedCertificate string
 	TrustedFingerprint string
 	NonInteractive     bool
 	NoAgent            bool
+	User               string
 }
 
-type loginMethod int8
-
-const (
-	localLogin loginMethod = iota
-	deviceLogin
-	oidcLogin
-)
-
-const cliLoginRedirectURL = "http://localhost:8301"
+const DeviceFlowMinVersion = "0.18.0"
 
 func newLoginCmd(cli *CLI) *cobra.Command {
 	var options loginCmdOptions
 
 	cmd := &cobra.Command{
-		Use:   "login [SERVER]",
-		Short: "Login to Infra",
-		Long:  "Login to Infra and start a background agent to keep local configuration up-to-date",
-		Example: `# By default, login will prompt for all required information.
-$ infra login
-
-# Login to a specific server
-$ infra login infraexampleserver.com
-
-# Login with a specific identity provider
-$ infra login --provider okta
-
-# Login with an access key
-$ export INFRA_ACCESS_KEY=1M4CWy9wF5.fAKeKEy5sMLH9ZZzAur0ZIjy
-$ infra login
-
-# Login with pre-set provider and server
-$ export INFRA_SERVER=example.infrahq.com
-$ export INFRA_PROVIDER=google
-$ infra login`,
+		Use:     "login [SERVER]",
+		Short:   "Login to Infra",
 		Args:    MaxArgs(1),
 		GroupID: groupCore,
+		Example: `
+# Login
+infra login example.infrahq.com
+
+# Login with username and password (prompt for password)
+infra login example.infrahq.com --user user@example.com
+
+# Login with access key
+export INFRA_SERVER=example.infrahq.com
+export INFRA_ACCESS_KEY=2vrEbqFEUr.jtTlxkgYdvghJNdEa8YoUxN0
+infra login example.infrahq.com --user user@example.com
+
+# Login with username and password
+export INFRA_SERVER=example.infrahq.com
+export INFRA_USER=user@example.com
+export INFRA_PASSWORD=p4ssw0rd
+infra login
+		`,
```

Comment:
Is this line intentional? It's add an empty line to the generated docs which isn't ideal.

Reply:
No – good catch!